src/screens/survey/checkbox.js

```diff
@@ -0,0 +1,119 @@
+import React from 'react';
+import { Text, View, Button,StyleSheet,TouchableHighlight,TouchableOpacity, Image, BackHandler} from 'react-native';
+
+class checkbox extends React.Component {
+
+  constructor(props){
+    super(props);
+    this.state = {
+      responseList : ["stronglyDisagree","disagree","noOpinion","agree","stronglyAgree"],
+      responseOption : {
+        stronglyDisagree : false,
+        disagree: false,
+        noOpinion: false,
+        agree: false,
+        stronglyAgree: false
+      },
+        previousResponse: null
+      }
+      this.toggleSelection = this.toggleSelection.bind(this)
+      this.startNewResponse = this.startNewResponse.bind(this)
+  }
+
+  componentDidMount(){
+    // Sets check to the Option previously selected
+    if(this.props.questionStatus['question'+this.props.index] != null){
+      this.toggleSelection(this.state.responseList[this.props.questionStatus['question'+this.props.index].response])
+    }
+  }
+
+  toggleSelection(currentResponse){
+    // A different way to set State might make this faster
+    var responseOption = {...this.state.responseOption}
+    if(this.state.previousResponse === null){
+      responseOption[currentResponse] = true;
+    } else {
+      responseOption[this.state.previousResponse] = false;
+      responseOption[currentResponse] = true;
+    }
+    this.setState({responseOption});
+    this.setState({previousResponse: currentResponse});
+  }
+
+  startNewResponse(currentSelection, response){
+    this.toggleSelection(currentSelection);
+    this.props.changeUserResponseField("questionResponse", response);
+    // There's a small delay that might be possible to fix with promises
+    setTimeout(() => this.props.nextScreen(), 5);
```

Comment:
What's causing the delay? Are you waiting for setState to resolve? If so, would it not be better to use a callback instead of a setTimeout. Would something like this work?

```
this.setState(prevState=> ({
      UserResponse: {
        ...prevState.UserResponse,
        [field]: value
      }
    }), () => this.nextScreen())
```

I tried this and was getting errors since this.props.navigation.navigate was expecting an object not this.state.index for it's second argument. Might it be better to use navigation state params over screenProps in this case? Besides resolving this error it would allow this code to be more explicit about what's being stored on screenProps. Thoughts?

Reply:
Yeah, so basically I'm waiting for changeUserResponseField to finish. I tried to use nagivation state params, but it didn't seem to be working. I might try to do something again with that though. At the time the only thing that worked was the screen props to pass parameters to all the screens.

Reply:
My guess is that navigating between screens is slow because you are re-rendering the entire navigation stack containing the questions each time you increment the index. It looks like your `SurveyMainScreenComponent` is primarily implemented to handle the business logic of navigating between screens and storing responses. I think it's an anti-pattern to do this. React components should only be concerned with rendering views, and Redux should handle the business logic.